docs/en/sdk-manual/web.md

```diff
@@ -70,167 +236,220 @@ Clickstream Web SDK supports the following data types:
-| _session_start_timestamp | Added in all events.                                                                                                                                       |
-| _session_duration        | Added in all events.                                                                                                                                       |
-| _session_number          | Added in all events. The initial value is 1, and the value is automatically incremented by user device.                                                    |
-| _error_code              | The `_clickstream_error` event's attribute.                                                                                                                |
-| _page_title              | Added in all events.                                                                                                                                       |
-| _page_url                | Added in all events.                                                                                                                                       |
+### Common attributes
+
+| Attribute name  | Data type | Description                                                       | How to generate                                                                                                                                                                                                                                            | Usage and purpose                                                                                     |
+|-----------------|-----------|-------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------|
+| hashCode        | string    | the event object's hash code                                      | calculated by library `@aws-crypto/sha256-js`                                                                                                                                                                                                              | distinguish different events                                                                          |
+| app_id          | string    | clickstream app id                                                | generated when clickstream app create from solution web console.                                                                                                                                                                                           | identify the events for your apps                                                                     |
```

Comment:
generated when clickstream app create from solution web console --> the app_id for the your app, app_id was generated by clickstream solution when you register an app to a data pipeline.

Reply:
fixed.